crates/sui/src/client_commands.rs

```diff
@@ -609,19 +614,25 @@ impl SuiClientCommands {
                         gas_budget,
                     )
                     .await?;
-                let signature =
-                    context
-                        .config
-                        .keystore
-                        .sign_secure(&sender, &data, Intent::default())?;
-                let response = context
-                    .execute_transaction(
-                        Transaction::from_data(data, Intent::default(), vec![signature])
-                            .verify()?,
-                    )
-                    .await?;
+                if serialize_output == false {
+                    let signature =
+                        context
+                            .config
+                            .keystore
+                            .sign_secure(&sender, &data, Intent::default())?;
+                    let response = context
+                        .execute_transaction(
+                            Transaction::from_data(data, Intent::default(), vec![signature])
+                                .verify()?,
+                        )
+                        .await?;
 
-                SuiClientCommandResult::Publish(response)
+                    SuiClientCommandResult::Publish(response)
+                } else {
+                    SuiClientCommandResult::SerializePublish(Base64::encode(
+                        bcs::to_bytes(&data).unwrap(),
+                    ))
+                }
             }
```

Comment:
just a code style nit here to avoid the `== false` -- could go for an early return instead?

```rust
if serialize_output {
    return SuiClientCommandResult::SerializePublish(Base64::encode(
        bcs::to_bytes(&data).unwrap(),
    ));
}

let signature =
    context
        .config
        .keystore
        .sign_secure(&sender, &data, Intent::default())?;

let response = context
    .execute_transaction(
        Transaction::from_data(data, Intent::default(), vec![signature])
            .verify()?,
    )
    .await?;

SuiClientCommandResult::Publish(response)
```